### PR TITLE
chore: polish APIScan and BinSkim scans

### PR DESCRIPTION
**Pull request by rzhao271**
Instead of integrating the CustomSDL stage into the Windows build, I believe it saves both pipeline run time and engineering time to keep the stage where it currently is. With the current stage, some compile steps are repeated, but they run in parallel to the rest of the pipeline. Because CustomSDL is isolated from the rest of the build, it also lowers the risk of accidentally publishing the product with symbols.

Pre-requisites before merging this PR:
- [x] Confirm whether electron.exe qualifies to be scanned, or whether we need to bring in Code.exe from a separate build process and scan that instead. @deepak1556 @rzhao271

### CODE REVIEW

**Comment by rzhao271** [comment]
Example run: https://dev.azure.com/monacotools/Monaco/_build/results?buildId=284637&view=results

**Comment by deepak1556** [comment]
Not a error from this PR but do you plan to address these errors in a follow-up ?

```
D:\a\_work\1\scanbin\VSCode-win32-x64\bin\ffmpeg.dll : error ERR997.ExceptionLoadingPdb : 'ffmpeg.dll' was not evaluated because its PDB could not be loaded (E_PDB_NOT_FOUND).
D:\a\_work\1\scanbin\VSCode-win32-x64\bin\libEGL.dll : error ERR997.ExceptionLoadingPdb : 'libEGL.dll' was not evaluated because its PDB could not be loaded (E_PDB_NOT_FOUND).
D:\a\_work\1\scanbin\VSCode-win32-x64\bin\libGLESv2.dll : error ERR997.ExceptionLoadingPdb : 'libGLESv2.dll' was not evaluated because its PDB could not be loaded (E_PDB_NOT_FOUND).
D:\a\_work\1\scanbin\VSCode-win32-x64\bin\vk_swiftshader.dll : error ERR997.ExceptionLoadingPdb : 'vk_swiftshader.dll' was not evaluated because its PDB could not be loaded (E_PDB_NOT_FOUND).
D:\a\_work\1\scanbin\VSCode-win32-x64\bin\vulkan-1.dll : error ERR997.ExceptionLoadingPdb : 'vulkan-1.dll' was not evaluated because its PDB could not be loaded (E_PDB_NOT_FOUND).
D:\a\_work\1\scanbin\VSCode-win32-x64\bin\electron.exe : error ERR997.ExceptionLoadingPdb : 'electron.exe' was not evaluated because its PDB could not be loaded (E_PDB_NOT_FOUND).

```

**Comment by deepak1556** [comment]
Have you explored running the `WindowsSDL` as part for the `Windows` stage as an independent job, that way you could reuse the node module cache and compilation cache. https://github.com/microsoft/vscode/pull/221592 explored it as a dependent job on `Windows` which has the issues you have pointed out so I am fine with not depending but atleast reuse some of the caches.

**Comment by rzhao271** [comment]
@deepak1556 I don't plan to address those errors until upstream follows up on https://github.com/microsoft/binskim/issues/661
Also, I tried using the node module cache in #221592 but I wasn't able to get it to work&mdash;the SHA kept on differing, resulting in cache misses.

**Comment by deepak1556** [comment]
Sorry ignore the node module cache since you are building with flags that are not in the original cache they will always miss. But you should still be able to use the compilation cache.

**Comment by rzhao271** [comment]
Is there a way to update the node modules cache? I believe that cache is much more important than the compilation output because APIScan and BinSkim only look at binary files, which are often located in node_modules. I also confirmed that the compilation output does not contain binaries.

In the meantime, could we merge this PR during debt week?

**Comment by deepak1556** [comment]
> Is there a way to update the node modules cache?

No we can't share the module cache in the current state unless the compilation flags between the current windows job and sdl job remains the same. So the question here is whether we can build the windows job with same flags as the sdl job.

> I believe that cache is much more important than the compilation output because APIScan and BinSkim only look at binary files, which are often located in node_modules. I also confirmed that the compilation output does not contain binaries.

Sharing the compilation cache would skip rebuilding the client stage and also makes one less step that is different from the windows job. Wouldn't it be better if more steps were shared across the windows job and the sdl job ?

> In the meantime, could we merge this PR during debt week?

I am fine with it, but please create a tracking issue to resolve the outstanding comments. Thanks!

File: build/azure-pipelines/win32/sdl-scan-win32.yml [hunk 1]
```diff
@@ -26,40 +25,34 @@ steps:
...
-      exec { npm config set registry "${{ parameters.NPM_REGISTRY }}" --location=project }
+      exec { npm config set registry "$env:NPM_REGISTRY" --location=project }
       # npm >v7 deprecated the `always-auth` config option, refs npm/cli@72a7eeb
       # following is a workaround for yarn to send authorization header
       # for GET requests to the registry.
       exec { Add-Content -Path .npmrc -Value "always-auth=true" }
-      exec { yarn config set registry "${{ parameters.NPM_REGISTRY }}" }
-    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'), ne('${{ parameters.NPM_REGISTRY }}', 'none'))
+      exec { yarn config set registry "$env:NPM_REGISTRY" }
+    condition: and(succeeded(), ne(variables.NODE_MODULES_RESTORED, 'true'), ne(variables['NPM_REGISTRY'], 'none'))
```
Comment (deepak1556) [inline build/azure-pipelines/win32/sdl-scan-win32.yml:41]: `NODE_MODULES_RESTORED` in this template doesn't seem right as the cache is never used today. Should remove dependency on it until the cache is restored.

**Review by deepak1556 — COMMENTED** [review]
LGTM with minor nit

**Review by deepak1556 — COMMENTED** [review]
LGTM, pending verification build

https://monacotools.visualstudio.com/Monaco/_build/results?buildId=286149&view=results

**Comment by rzhao271** [comment]
I have verified that the Windows x64 user setup is sane and that the archive does not contain PDB files.

**Review by deepak1556 — DISMISSED** [review]

**Comment by rzhao271** [comment]
Verification build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=286891&view=results

**Review by deepak1556 — APPROVED** [review]

**Pull request merged**